src/JSTest.Runner/TestRunner/TestFrameworks/Jest/JestTestFramework.ts

```diff
@@ -69,38 +69,77 @@ export class JestTestFramework extends BaseTestFramework {
     }
 
     public startExecutionWithTests(sources: Array<string>, testCollection: Map<string, TestCase>, options: JSON) {
-        let runConfigPath;
-        try {
-            // tslint:disable-next-line:no-string-literal
-            runConfigPath = testCollection.entries().next().value[1].Properties[0]['Value'];
-        } catch (e) {
-            throw new Exception('TestCase object does not contain jestConfigPath in Properties', ExceptionType.TestFrameworkError);
+        const configToSourceMap: Map<string, Array<string>> = new Map();
+        const testNames = [];
+
+        const testCaseIterator = testCollection.values();
+        let testCaseIteration = testCaseIterator.next();
+        while (!testCaseIteration.done) {
+            const testCase = testCaseIteration.value;
+            const fqnRegex = testCase.FullyQualifiedName.match(/.*::(.*)::(.*)/);
+
+            const configPath = testCase.Source;
+
+            if (fqnRegex) {
+                // source path appended to the fqn is relative to the config file
+                testNames.push(fqnRegex[1]);
+
+                const source = path.normalize(path.dirname(configPath) + '\\' + fqnRegex[2]);
+                if (configToSourceMap.has(configPath)) {
+                    configToSourceMap.get(configPath)[source] = 1;
+                } else {
+                    const sourceArray = [];
+                    sourceArray[source] = 1;
+                    configToSourceMap.set(configPath, sourceArray);
+                }
+            } else {
+                // console.warn('Config file not provided in fqn for source:', src);
```

Comment:
>// console.warn('Config file not provided in fqn for source:', src); [](start = 16, length = 68)

better do it.. not sure if you will land up here..

Reply:
please test with the PR..

---
In reply to: [245627010](https://github.com/karanjitsingh/jstestadapter/pull/11#discussion_r245627010) [](ancestors = 245627010)